openedx/core/djangoapps/course_groups/views.py

```diff
@@ -90,7 +90,8 @@ def _get_cohort_representation(cohort, course):
     return {
         'name': cohort.name,
         'id': cohort.id,
-        'user_count': cohort.users.count(),
+        'user_count': cohort.users.filter(courseenrollment__course_id=course.location.course_key,
```

Comment:
Please add a unit test to cover this. You can probably build on to CohortHandlerTestCase in test_views.py.